lib/policy_machine.rb

```diff
@@ -95,26 +95,39 @@ def is_privilege?(user_or_attribute, operation, object_or_attribute, options = {
   end
 
   ##
-  # Check the privilege without checking for prohibitions. May be called directly but is also used in is_privilege?
-  #
-  def is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute, options = {})
-    unless user_or_attribute.is_a?(PM::User) || user_or_attribute.is_a?(PM::UserAttribute)
-      raise(ArgumentError, "user_attribute_pe must be a User or UserAttribute.")
-    end
+  # Can we derive a privilege given a filtered set of user attributes?
+  def is_privilege_with_filters?(user_or_attribute, operation, object_or_attribute, filters: {}, options: {})
+    is_privilege_ignoring_prohibitions_with_filters?(user_or_attribute, operation, object_or_attribute, filters: filters, options: options) && (options[:ignore_prohibitions] || !is_privilege_ignoring_prohibitions?(user_or_attribute, PM::Prohibition.on(operation), object_or_attribute, options))
+  end
 
-    unless [PM::Operation, Symbol, String].any? { |allowed_type| operation.is_a?(allowed_type) }
-      raise(ArgumentError, "operation must be an Operation, Symbol, or String.")
-    end
+  ##
+  # Check the privilege with filters without checking for prohibitions.
+  def is_privilege_ignoring_prohibitions_with_filters?(user_or_attribute, operation, object_or_attribute, filters: {}, options: {})
```

Comment:
So for these PM level public interfaces, I think it would be worthwhile to represent that we're only doing UA filtering.  I think we could either 1) have user attribute specific filtering method names or 2) make the filter args more complicated, so expect keys like `filters: {user_attributes: {key: "val"}}`, which would also allow backwards compatible expansion in future

Reply:
Agreed, I think making the filter args more specific is the best path going forward; fixed